From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Tue, 29 Nov 2022 01:15:55 +0100
Subject: [PATCH] Hide irrelevant compilation warnings

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

This patch is based on the following patch:
"Hide irrelevant compilation warnings"
By: Martijn Muijsers <martijnmuijsers@live.nl>
As part of: Gale (https://github.com/GaleMC/Gale)
Licensed under: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)

* Gale description *

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Gale - https://galemc.org

diff --git a/build.gradle.kts b/build.gradle.kts
index 149f9088fe806467656e8b1c4157df60fda69ba7..53ac9109124589d464d11e1517b742a4cf6665e9 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -89,6 +89,15 @@ val generateApiVersioningFile by tasks.registering {
     }
 }
 
+// Fiddle start - Gale - hide irrelevant compilation warnings
+tasks.withType<JavaCompile> {
+    val compilerArgs = options.compilerArgs
+    compilerArgs.add("-Xlint:-module")
+    compilerArgs.add("-Xlint:-removal")
+    compilerArgs.add("-Xlint:-dep-ann")
+}
+// Fiddle end - Gale - hide irrelevant compilation warnings
+
 tasks.jar {
     from(generateApiVersioningFile.map { it.outputs.files.singleFile }) {
         into("META-INF/maven/${project.group}/${project.name}")
@@ -141,6 +150,8 @@ tasks.withType<Javadoc> {
             into("build/docs/javadoc")
         }
     }
+
+    options.addStringOption("Xdoclint:none", "-quiet") // Fiddle - Gale - hide irrelevant compilation warnings
 }
 
 // Paper start
